### PR TITLE
fix: update Go Encryption compatibility

### LINKED ISSUE #196: Verify the Go JWE Encryption with the proposed JWE PHP library
As part of the Epic issue #36,

the Go JWE encryption implementation must match the proposed encryption build in the  PHP library: https://github.com/gamringer/php-authcrypt/blob/master/src/Crypt.php

Encryption made Aries-framework-go must generate the same result if using the same keys as in the PHP link above

### PR DESCRIPTION
**Pull request by Baha-sk**
closes #196

### CODE REVIEW

File: pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go [hunk 1]
```diff
@@ -434,80 +434,22 @@ func TestBadCreateCipher(t *testing.T) {
...
-    "iv": "nM0UDDCERek5syITAHwzDPGiEVErTtpo",
-    "tag": "gu1ZvF35-JYMd1JITD0qeg",
-    "ciphertext": "ntZwQokGaZhnQ8L2"
-}`
-
-	senderKp := jwecrypto.KeyPair{
-		Priv: senderPriv,
-		Pub:  senderPub,
-	}
+	const refJWE = `{"protected":"eyJ0eXAiOiJwcnMuaHlwZXJsZWRnZXIuYXJpZXMtYXV0aC1tZXNzYWdlIiwiYWxnIjoiRUNESC1TUytYQzIwUEtXIiwiZW5jIjoiWEMyMFAifQ","recipients":[{"encrypted_key":"-FqJHyCb52lPTU0GkjTLFHPv-bIjk7zV7j3yQmcUuaU","header":{"apu":"EbofDL0elLBv4pcoeyjo_tgx3Cd0SA6NqW5qWPAqU--7RTWKosW_7YzrA2KbbpEcAOCBv-GWoa9Hyrsuil_luQ","iv":"D5-segFndIlJRLAwSoFHKjtItV2DHSHr","tag":"wllnsUZ0LD1ktiu1dZoXWQ","kid":"4n2AqYZ9kzyTWWZJnS6wSuhkqzW9qUdQHpoWXSBG6KwR","oid":"Q_DA4thb6f5_IyZOTYQJjQHor336K3kb2erwfRqq6xmrHZTwv1OgwxikJErpDqJ6t-Cf27N2IWyvcV6Z0Z0a3322AZQmfV2nMXgyCWV42ad395jIE9zpPSvi6CE"}}],"aad":"jzdqnBug3dzkhEyWVzHA-JhCS7OHkVNxMp4DxVVxPAE","iv":"TEk7JtQZwsU0mBJeew9DH2npmxjr8_Mr","tag":"zFMdWklZaxCymIEPO_5Cuw","ciphertext":"WgRFQZJ_43UwtRQm"}` //nolint:lll
```
Comment (troyronda) [inline pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go]: This line could be formatted like it was before. Also is this change to ref JWE needed?

File: pkg/didcomm/crypto/jwe/authcrypt/decrypt.go [hunk 2]
```diff
@@ -134,11 +120,6 @@ func (c *Crypter) findRecipient(jweRecipients []Recipient, recipientPubKey *[cha
 }
 
 func (c *Crypter) decryptSharedKey(recipientKp jwecrypto.KeyPair, senderPubKey *[chacha.KeySize]byte, recipient *Recipient) ([]byte, error) { //nolint:lll
-	apu, err := base64.RawURLEncoding.DecodeString(recipient.Header.APU)
```
Comment (troyronda) [inline pkg/didcomm/crypto/jwe/authcrypt/decrypt.go]: This function doesn’t look like it was actually changed in logic.

File: pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go (hunk 1 above)
Comment (Baha-sk) [reply pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go]: older format had a linter error (lll) that I can't comment inside of a multi-line string

File: pkg/didcomm/crypto/jwe/authcrypt/decrypt.go (hunk 2 above)
Comment (Baha-sk) [reply pkg/didcomm/crypto/jwe/authcrypt/decrypt.go]: sure, I can pull up apu back here to match this code..

File: pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go (hunk 1 above)
Comment (troyronda) [reply pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go]: You put the no lint comment in the line before the const.

Comment (Baha-sk) [reply pkg/didcomm/crypto/jwe/authcrypt/authcrypt_test.go]: done

File: pkg/didcomm/crypto/jwe/authcrypt/decrypt.go (hunk 2 above)
Comment (Baha-sk) [reply pkg/didcomm/crypto/jwe/authcrypt/decrypt.go]: I pulled apu DecodeString to its original place

**Comment by Baha-sk** [comment]
> Semantic title should be included.

updated to `fix:`

**Pull request merged**